project/schema_base.py

```diff
@@ -276,6 +276,29 @@ def perform_mutate(cls, form: forms.LogoutForm, info: ResolveInfo):
         return cls.mutation_success()
 
 
+@schema_registry.register_mutation
+class ClearAnonymousSession(SessionFormMutation):
+    """
+    Clears the session, but only if the user is not logged in; if they are logged in,
+    this does nothing.
+
+    Clients should pay attention to the CSRF token, because apparently this changes
```

Comment:
Maybe something a little more specific about what clients have to do would be helpful here. Do they have to make sure it gets reset? Do they have to make sure it stays the same?

Reply:
They just need to make sure it stays "synced" with whatever the server says its current state is: if the server says it's  changed, then the front-end needs to update itself so that it uses the new one whenever it makes network requests, and if it stays the same, then it doesn't need to do anything.  This is all generally done automatically by the [`SessionUpdatingFormSubmitter`](https://github.com/JustFixNYC/tenants2/blob/master/frontend/lib/forms/session-updating-form-submitter.tsx) component on the front-end, as long as the GraphQL query for the mutation explicitly mentions `session.csrfToken` as a return value of the mutation it wants to know about (which it does here).

I've updated this documentation in a commit I'm about to push, so hopefully that should make it a little less confusing.